pkg/seedadmissioncontroller/webhooks/admission/extensionresources/admission.go

```diff
@@ -16,261 +16,266 @@ package extensionresources
-			},
-		},
-
-		gvr("infrastructures"): {
-			newObject: func() client.Object { return new(extensionsv1alpha1.Infrastructure) },
-			validateCreateResource: func(n, _ client.Object) field.ErrorList {
-				return validation.ValidateInfrastructure(n.(*extensionsv1alpha1.Infrastructure))
-			},
-			validateUpdateResource: func(n, o client.Object) field.ErrorList {
-				return validation.ValidateInfrastructureUpdate(n.(*extensionsv1alpha1.Infrastructure), o.(*extensionsv1alpha1.Infrastructure))
-			},
-		},
-
-		gvr("networks"): {
-			newObject: func() client.Object { return new(extensionsv1alpha1.Network) },
-			validateCreateResource: func(n, _ client.Object) field.ErrorList {
-				return validation.ValidateNetwork(n.(*extensionsv1alpha1.Network))
-			},
-			validateUpdateResource: func(n, o client.Object) field.ErrorList {
-				return validation.ValidateNetworkUpdate(n.(*extensionsv1alpha1.Network), o.(*extensionsv1alpha1.Network))
-			},
-		},
-
-		gvr("operatingsystemconfigs"): {
-			newObject: func() client.Object { return new(extensionsv1alpha1.OperatingSystemConfig) },
-			validateCreateResource: func(n, _ client.Object) field.ErrorList {
-				return validation.ValidateOperatingSystemConfig(n.(*extensionsv1alpha1.OperatingSystemConfig))
-			},
-			validateUpdateResource: func(n, o client.Object) field.ErrorList {
-				return validation.ValidateOperatingSystemConfigUpdate(n.(*extensionsv1alpha1.OperatingSystemConfig), o.(*extensionsv1alpha1.OperatingSystemConfig))
-			},
-		},
-
-		gvr("workers"): {
-			newObject: func() client.Object { return new(extensionsv1alpha1.Worker) },
-			validateCreateResource: func(n, _ client.Object) field.ErrorList {
-				return validation.ValidateWorker(n.(*extensionsv1alpha1.Worker))
-			},
-			validateUpdateResource: func(n, o client.Object) field.ErrorList {
-				return validation.ValidateWorkerUpdate(n.(*extensionsv1alpha1.Worker), o.(*extensionsv1alpha1.Worker))
-			},
-		},
+// AddWebhooks add extension's valdation webhook to manager
+func AddWebhooks(mgr manager.Manager) error {
+	for obj, validator := range validators {
+		if err := builder.WebhookManagedBy(mgr).WithValidator(validator).For(obj).Complete(); err != nil {
+			return err
+		}
 	}
 
-	h := handler{
-		logger:    logger,
-		artifacts: artifacts,
+	return nil
+}
+
+var (
+	_ admission.CustomValidator = &backupBucketValidator{}
+	_ admission.CustomValidator = &backupEntryValidator{}
+	_ admission.CustomValidator = &bastionValidator{}
+	_ admission.CustomValidator = &containerRuntimeValidator{}
+	_ admission.CustomValidator = &controlPlaneValidator{}
+	_ admission.CustomValidator = &dnsRecordValidator{}
+	_ admission.CustomValidator = &etcdValidator{}
+	_ admission.CustomValidator = &extensionValidator{}
+	_ admission.CustomValidator = &infrastructureValidator{}
+	_ admission.CustomValidator = &networkValidator{}
+	_ admission.CustomValidator = &operatingSystemConfigValidator{}
+	_ admission.CustomValidator = &workerValidator{}
```

Comment:
You can drop these, no need to enforce interface compliance again, as it's already enforced by other calls and the map below.